### PR TITLE
Added DPS value type configuration

### PR DESCRIPTION
**Pull request by deese**
This PR does the following:

1. Adds a configuration option on the type of data sent to the fan DP.  
2. Fixes the error when using an ordered list.

**[ 1 ] Reasoning**

In my brand (Ikohs) the set_dp call needs to send an integer number if not, the fan starts making some weird noises and the speed is not properly changed. If an integer is sent, this behaviour doesn´t happen. This means that some brands requires an integer and others may require an string, therefore, I've added the type of data to be sent as an option, so the user can change it when configuring the entity.

image

**[ 2 ] Reasoning**

Seems that the data stored in the list is different from the one being checked so the call fails. This PR fixes this by casting the current_speed variable to string when using an ordered_list to convert from ordered list to percentage (ordered_list_item_to_percentage)

2022-05-24 15:17:28 ERROR (SyncWorker_7) [homeassistant.util.logging] Exception in _update_handler when dispatching 'localtuya_xxxx': ({'20': False, '22': 1000, '23': 0, '60': True, '62': 3, '63': 'forward', '64': 0},)
Traceback (most recent call last):
File "/config/custom_components/localtuya/common.py", line 281, in _update_handler
self.status_updated()
File "/config/custom_components/localtuya/fan.py", line 223, in status_updated
self._percentage = ordered_list_item_to_percentage(:
File "/usr/src/homeassistant/homeassistant/util/percentage.py", line 25, in ordered_list_item_to_percentage
raise ValueError(f'The item "{item}"" is not in "{ordered_list}"')
ValueError: The item "3"" is not in "['1', '2', '3', '4', '5', '6']"

### CODE REVIEW

**Review by CloCkWeRX — APPROVED** [review]

**Comment by CloCkWeRX** [comment]
I think this is a more comprehensive approach than #823

**Review by joaquinvacas — APPROVED** [review]

**Review by KardinalReusen — APPROVED** [review]

**Review by joaquinvacas — COMMENTED** [review]
Works fine, tested in 3 different Tuya ceiling fans

**Comment by CloCkWeRX** [comment]
@deese want to rebase 1x more time, adjust the tox complaint?

custom_components/localtuya/fan.py:52:80: W291 trailing whitespace

**Comment by CloCkWeRX** [comment]
@rospogrigio this seems like a good candidate (there are a few other PRs that either add, remove, etc the DPS type between string and int)

**Comment by rospogrigio** [comment]
Yes, please fix the trailing whitespace issue and then let's merge it.

File: custom_components/localtuya/fan.py [hunk 1]
```diff
@@ -48,6 +49,7 @@ def flow_schema(dps):
         vol.Optional(CONF_FAN_SPEED_MIN, default=1): cv.positive_int,
         vol.Optional(CONF_FAN_SPEED_MAX, default=9): cv.positive_int,
         vol.Optional(CONF_FAN_ORDERED_LIST, default="disabled"): cv.string,
+        vol.Optional(CONF_FAN_DPS_TYPE, default='str'): vol.In(["str", "int"]),
```
Comment (CloCkWeRX) [inline custom_components/localtuya/fan.py:52]:
```suggestion
        vol.Optional(CONF_FAN_DPS_TYPE, default='str'): vol.In(["str", "int"]),
```

Comment (CloCkWeRX) [reply custom_components/localtuya/fan.py:52]: @deese - bump! This one is pretty darned close / ready to go.

**Comment by CloCkWeRX** [comment]
I think... that's ready now @rospogrigio :)

**Comment by joaquinvacas** [comment]
Sorry but, does the new "Tuya number" fixes available in new versions fix this?

**Comment by KardinalReusen** [comment]
Will the value type configuration be present in the next release?

**Comment by ingoratsdorf** [comment]
This fix should not only go into fans. This is also required for other types, like PIR and countdown, basically anywhere requiring an int and not a float. Ref https://github.com/rospogrigio/localtuya/issues/931

**Comment by vampywiz17** [comment]
Any idea when will be merge this PR? My fan controllers are not working now.

**Comment by rospogrigio** [comment]
> Any idea when will be merge this PR? My fan controllers are not working now.

When the PR owner will fix the failing check...

**Comment by vampywiz17** [comment]
> > Any idea when will be merge this PR? My fan controllers are not working now.
> 
> When the PR owner will fix the failing check...

I see. i wait.

**Comment by deese** [comment]
> > Any idea when will be merge this PR? My fan controllers are not working now.
> 
> When the PR owner will fix the failing check...

totally overlooked that one. Checks passing now :)

**Review by rospogrigio — APPROVED** [review]

**Comment by rospogrigio** [comment]
PR merged.

**Pull request merged**